specs/select.spec.md

```diff
@@ -0,0 +1,156 @@
+- `required` - boolean value that sets the field as required
+- `selectedIndex` - The index of the first or last selected `<fast-option>` element, depending on the value of `multiple`
+- `selectedOptions` - An `HTMLCollection` of the selected `<fast-option>` elements
+- `size` - If the `<fast-select>` is shown with a scrollbar, this represents how many `<fast-option>`s are visible.
+- `value` - string value of the number field, can be an empty string
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the `selectedIndex`'s value change.
+
+*Slots*
+- default - the list of option(s)
+- content - the content of the button
+- start - often times a glyph, icon, or button precedes the content
+- end - often times a glyph, icon, or button follows the content
+- expand-glyph - glyph that indicates the select is expandable, often represented by a downward chevron.
+
+### Anatomy and Appearance
+**Structure:**
+
+```html
+  <template>
+    <button
+          part="button"
+          class="button"
+    >
+        <slot name="start" part="start"></slot>
+        <slot name="content" part="content"></slot>
+        <slot name="end" part="end"></slot>
+        <slot name="expand-glyph" part="expand-glyph"></slot>
+    </button>
+    <ul
+        part="listbox"
+        class="listbox"
+    >
+        <slot></slot>
+    </ul>
+  </template>
+```
+
+## Implementation
+
+```html
+<fast-select>
+    <fast-option>Option 1</fast-option>
+    <fast-option>Option 2</fast-option>
+    <fast-option>Option 3</fast-option>
+    <fast-option>Option 4</fast-option>
+    <fast-option>Option 5</fast-option>
+</fast-select>
+```
+
+### States
+
+**open**: This state is applied to the `<fast-select>` when the listbox is visible to the user.
+
+**required**: An `<fast-option>` from the `<fast-select>` must be selected when the `required` attribute is set to `true`
+
+**valid**: The `<fast-select>` meets all its validation constraints, and is therefore considered to be valid.
+
+**invalid**: The `<fast-select>` does not meet its validation constraints, and is therefore considered to be invalid.
+
+### Accessibility
+
+*Select* are RTL compliant and support the following aria best practices for listbox [W3C aria-practices](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox)
+
+### Test Plan
+
+While testing is still TBD for our web components, I would expect this to align with the testing strategy and not require any additional test support.
+
+# Select Option
+
+## Overview
+
+The `option` component is an option that is meant to be used with `<fast-select>`. `option` renders inside the `<fast-select>` listbox dropdown.
+
+### API
+*Component Name*:
+- `fast-option`
+
+*Attributes:*
+- `defaultSelected` - Sets the default of the `<fast-option>` to selected.
+- `disabled` - disables the control
+- `selected` - If the `<fast-option>` is selected or not.
+- `value` - string value of the number field, can be an empty string
```

Comment:
Looks like a copy/paste here "number field".

Also, I think it would be valuable for `value` to allowed to be any type when accessed through the property. Attributes have to be strings, but the property doesn't. This would allow templating over lists of complex objects, and associating the complex object as the value if desired.

Reply:
D'oh was following an example of a form associated component a bit too closely, ;)